camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlMultiOutputCommitter.java

```diff
@@ -188,6 +177,31 @@ public void commitTask(TaskAttemptContext context) throws IOException {
     super.commitTask(context);
   }
 
+  private void uploadToGcs(TaskAttemptContext context, FileSystem fs, String partitionedFile, Path dest) {
+    for (int retries = 0; retries <= MAX_GCS_UPLOAD_RETRIES; retries++) {
+      try {
+        Configuration googleConfig = new Configuration(fs.getConf());
+        googleConfig.set("fs.default.name", EtlMultiOutputFormat.getGCSPrefix(context));
+        // FileSystem.newInstace could throw IOException which is why I'm keeping
+        // all of this setup inside the try block despite being potentially wasteful.
+        FileSystem googleFs = FileSystem.newInstance(googleConfig);
+        Path baseOutDirGCS = EtlMultiOutputFormat.getDestinationPathGCS(context);
+        Path gcsDest = new Path(baseOutDirGCS, partitionedFile);
+        // copy the file that we've committed to gcs
+        uploadFile(fs, dest, googleFs, gcsDest, googleConfig);
+        return;
+      } catch (Exception e) {
+        if (retries == MAX_GCS_UPLOAD_RETRIES - 1) {
```

Comment:
I think this should be `if (retries < MAX_GCS_UPLOAD_RETRIES)`, or equivalently, only do the else block if `(retries == MAX_GCS_UPLOAD_RETRIES)`

Reply:
ah of course. Thanks!